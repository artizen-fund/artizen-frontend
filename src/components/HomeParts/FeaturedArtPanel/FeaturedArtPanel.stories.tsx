import FeaturedArtPanel from './'

export default {
  title: 'components/FeaturedArtPanel',
  component: FeaturedArtPanel,
  argTypes: {},
}

export const FeaturedArtPanelComponent = (props: any) => <FeaturedArtPanel {...props} />
