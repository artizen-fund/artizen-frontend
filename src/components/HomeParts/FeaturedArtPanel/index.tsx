import styled from 'styled-components'
import { Icon, TableAvatar } from '@components'
import { rgba, formatDate } from '@lib'
import { palette, typography } from '@theme'
import { IGrantsWithProjectFragment } from '@types'

type IFeaturedArtPanel = {
  grant?: IGrantsWithProjectFragment
  loading: boolean
}

const FeaturedArtPanel = ({ grant }: IFeaturedArtPanel) => {
  const artist = grant?.submission?.project?.members?.filter(m => m.type === 'lead')[0]

  return !grant ? (
    <></>
  ) : (
    <Wrapper>
      <Copy>
        <Metadata>
          <Metadatum>
            <Icon glyph="crown" color="slate" darkColor="moon" level={2} outline glyphOutline label="Top Donor Prize" />
          </Metadatum>
          <Metadatum>
            <Icon glyph="palette" color="slate" darkColor="moon" level={2} outline glyphOutline label="Artifact 27" />
          </Metadatum>
          <Metadatum>
            <Icon
              glyph="face"
              color="slate"
              darkColor="moon"
              level={2}
              outline
              glyphOutline
              label={`${artist?.user?.firstName} ${artist?.user?.lastName}`}
            />
          </Metadatum>
        </Metadata>
      </Copy>
      <P>{grant.submission?.project?.description}</P>
      <Impact>Impact</Impact>
      <P>{grant.submission?.project?.impact}</P>

      <ListHeader>Project</ListHeader>
      <List>
        <div>
          <dt>Season {grant.season}</dt>
          <dd>Extended Reality</dd>
        </div>
        <div>
          <dt>Started</dt>
          <dd>{formatDate(grant.submission?.project?.creationDate)}</dd>
        </div>
        <div>
          <dt>Completed</dt>
          <dd>{formatDate(grant.submission?.project?.completionDate)}</dd>
        </div>
      </List>

      {/* todo: waiting on @EK for data
      <ListHeader>Artifact</ListHeader>
      <List>
        <div>
          <dt>Minted</dt>
          <dd>createdAt</dd>
        </div>
        <div>
          <dt>Token</dt>
          <dd>{grant.submission?.artifacts[0]?.blockchainAddress}</dd>
        </div>
      </List>
      */}

      <ListHeader>Contributors</ListHeader>
      <List>
        {grant?.submission?.project?.members.map((member, index) => (
          <div key={`member-${index}`}>
            <dt>
              <TableAvatar profileImage={member.user?.profileImage} />
              {member?.user?.firstName} {member?.user?.lastName}
            </dt>
            <dd>{member.type}</dd>
          </div>
        ))}
      </List>
    </Wrapper>
  )
}

const Wrapper = styled.div`
  grid-area: tabbedInfo;
  margin-top: 30px;
`

const Copy = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
`

const Title = styled.div`
  ${typography.title.l4}
  margin: 1em 0;
`

const Metadata = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  gap: 10px 15px;
  ${typography.label.l1}
  margin-bottom: 24px;
`

const Metadatum = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  gap: 10px;
`

const P = styled.p`
  ${typography.body.l2}
`

const Impact = styled.h4`
  margin-top: 1em;
  ${typography.title.l4}
`

const ListHeader = styled.h4`
  ${typography.label.l1}
  margin: 1em 0;
`

const List = styled.dl`
  ${typography.label.l1}
  display: flex;
  flex-direction: column;
  gap: 4px;
  > div {
    background: ${rgba(palette.stone, 0.24)};
    border-radius: 4px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    height: 40px;
    padding: 0 15px;
    dt {
      display: flex;
      flex-direction: row;
      align-items: center;
      gap: 15px;
    }
  }
`

export default FeaturedArtPanel
