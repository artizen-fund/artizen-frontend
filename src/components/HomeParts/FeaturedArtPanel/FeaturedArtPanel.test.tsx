import renderer from 'react-test-renderer'
import FeaturedArtPanel from './'

describe('FeaturedArtPanel', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<FeaturedArtPanel />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
