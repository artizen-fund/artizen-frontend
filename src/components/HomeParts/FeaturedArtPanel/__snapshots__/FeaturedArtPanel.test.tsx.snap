// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedArtPanel renders correctly 1`] = `
<div
  className="sc-b73dc9bf-1 lmHMNT"
>
  <div
    className="sc-b73dc9bf-2 dMuaEB"
  >
    <div
      className="sc-b73dc9bf-4 dnUafx"
    >
      <div
        className="sc-b73dc9bf-5 hSnUwu"
      >
        <div
          className="sc-b51f94be-2 dSzGHa"
        >
          <div
            className="sc-3630b030-0 btCMnq sc-b51f94be-3 fcATQK"
          >
            <div
              className="sc-3630b030-1 iydzSb"
            />
          </div>
          <div
            className="sc-3630b030-0 btCMnq"
          >
            <div
              className="sc-3630b030-1 iydzSb"
            />
          </div>
        </div>
      </div>
      <div
        className="sc-b73dc9bf-5 hSnUwu"
      >
        <div
          className="sc-b51f94be-2 dSzGHa"
        >
          <div
            className="sc-3630b030-0 btCMnq sc-b51f94be-3 fcATQK"
          >
            <div
              className="sc-3630b030-1 iydzSb"
            />
          </div>
          <div
            className="sc-3630b030-0 btCMnq"
          >
            <div
              className="sc-3630b030-1 iydzSb"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-3630b030-0 btCMnq"
  >
    <div
      className="sc-3630b030-1 iydzSb"
    />
  </div>
  <div
    className="sc-b73dc9bf-9 lcEMcE"
  >
    <div
      className="react-slidedown sc-b73dc9bf-0"
      onTransitionEnd={[Function]}
    >
      <h4
        className="sc-b73dc9bf-8 gpEyl"
      >
        Project
      </h4>
      <dl
        className="sc-b73dc9bf-10 jaMSBY"
      >
        <div>
          <dt>
            Started
          </dt>
          <dd>
            undefined NaN, NaN
          </dd>
        </div>
        <div>
          <dt>
            Completed
          </dt>
          <dd>
            undefined NaN, NaN
          </dd>
        </div>
      </dl>
      <h4
        className="sc-b73dc9bf-8 gpEyl"
      >
        Artifact
      </h4>
      <dl
        className="sc-b73dc9bf-10 jaMSBY"
      >
        <div>
          <dt>
            Season 
          </dt>
          <dd>
            Genesis
          </dd>
        </div>
        <div>
          <dt>
            Minted
          </dt>
          <dd>
            undefined NaN, NaN
          </dd>
        </div>
      </dl>
      <h4
        className="sc-b73dc9bf-8 gpEyl"
      >
        Contributors
      </h4>
      <dl
        className="sc-b73dc9bf-10 jaMSBY"
      />
      <button
        className="sc-313ded62-2 bSEyzZ sc-b73dc9bf-11 epmAqQ "
        onClick={[Function]}
      >
        <span>
          view less
        </span>
      </button>
    </div>
  </div>
</div>
`;
